### DISCUSSION TITLE
PA Engine used in SaaS and tm1py

### DISCUSSION THREAD

**Discussion by janus-t** [cubewise-code/tm1py#962] [Q&A] [answered]
IBM introduced some time ago a new Planning Analytics version that they provide as SaaS. Have you had change to look into new version and do you think tm1py could be used with it as well?

It seems that this version has different authentication method and REST API url structure. In my tests authentication is based on apikey instead of familiar version 11 REST API authentication modes. Another thing that I noted is that api url is different and if I checked right, tm1py has url = '/api/v1/<command>' hardcoded into request functions.

**Comment by MariusWirtz** [comment]
Here is a sample of how to use TM1py with TM1 (v11) in PAoC

``` python
with TM1Service(
        base_url='https://hostname.planning-analytics.ibmcloud.com/tm1/api/tm1_database_name/',
        user="user",
        namespace="LDAP",
        password="xxxx",
        ssl=True,
        verify=True,
        async_requests_mode=True) as tm1:
    print(tm1.server.get_product_version())
```

But if I understand you right, you are looking for a version of TM1py that is compatible with what we call TM1 v12.
@rclapp started working on this feature here: https://github.com/rclapp/tm1py/tree/tm1-v12-ga-release-testing
This feature will almost certainly be included in the next major release of TM1py.

**Reply by janus-t** [reply]
You got it. I was talking about this TM1 v12. I just didn't know what to call it. IBM documentation calls it PA Engine in many places.

Great to hear that you are working on it!

**Comment by rclapp** [comment, marked as answer]
You can track the progress to support SaaS aka v12 here: https://github.com/cubewise-code/tm1py/issues/889
